workflow/installertest.groovy

```diff
@@ -92,41 +205,79 @@ def runShellTest(String imageName, def shellCommands, def stepNames=null) {
       }
+      // And archive the test results once we're done.
+      step([$class: 'JUnitResultArchiver', keepLongStdio: true, testResults: "results/*.xml"])
       archive("testresults/$fileName"+'*.log')
     }
   } 
 }
 
 
-/** Install tests are a set of ["dockerImage:version", [shellCommand,shellCommand...]] entries
-* They will need sudo-able containers to install
-* @param stepNames Names for each step (if not supplied, the index of the step will be used)
+/**
+ * Install tests are a set of ["dockerImage:version", [shellCommand,shellCommand...]] entries
+ * They will need sudo-able containers to install
+ *
+ * @param testOSList a list of OS information (filtered from osesToTest(category))
+ * @param scriptPath the path to the scripts for replacing
+ * @param artifactName The name of the artifact to test
+ * @param jenkinsPort The port to run the test Jenkins on
+ * @param stepNames Names for each step (if not supplied, the index of the step will be used)
 */
-def executeInstallTestset(def coreTests, def stepNames=null) {
+def executeInstallTestset(List testOSList, String scriptPath, String artifactName, String jenkinsPort, def stepNames=null) {
   // Within this node, execute our docker tests
   def parallelTests = [:]
   sh 'rm -rf testresults'
+  sh 'rm -rf results'
   sh 'mkdir testresults'
-  for (int i=0; i<coreTests.size(); i++) {
-    def imgName = coreTests[i][0]
-    def tests = coreTests[i][1]
-    parallelTests[imgName] = {
-      try {
-       runShellTest(imgName, tests, stepNames)
-      } catch(Exception e) {
-        // Keep on trucking so we can see the full failures list
-        echo "$e"
-        error("Test for $imgName failed")
+  sh 'mkdir results'
+  for (int i = 0; i < testOSList.size(); i++) {
+    def osDef = testOSList.get(i)
+    if (osDef != null) {
+      // OS name is index 0 element.
+      def osName = osDef.get(0)
+      // Image name is index 1 element.
+      def imgName = osDef.get(1)
+      // OS Platform is index 2 element.
+      def osPlatform = osDef.get(2)
+
+      def tests = []
+
+      def scriptsToRun = []
+      scriptsToRun.add(platformScripts(osPlatform))
+      def genericScripts = standardScripts(artifactName, jenkinsPort)
+
+      for (int k = 0; k < genericScripts.size(); k++) {
+        scriptsToRun.add(genericScripts.get(k))
+      }
+
+      for (int j = 0; j < scriptsToRun.size(); j++) {
+        def scriptToRun = scriptsToRun.get(j)
+        // Since we can't use Maps due to serialization fun, we're faking it with lists, so scriptToRun[0] is the "platform", while
+        // scriptToRun[1] and [2] are the script name and arguments, respectively.
+        tests << shellScriptForDistro(osName, scriptPath, scriptToRun[1], scriptToRun[2])
+      }
+
+      parallelTests[osName] = {
+        try {
+          runShellTest(imgName, tests, stepNames)
+        } catch (Exception e) {
+          // Keep on trucking so we can see the full failures list
+          echo "${e}"
```

Comment:
I'm confused - in this scenario, at least, we don't *want* to stop the build if it's unstable, we want to keep going and make sure we run all the tests for all the platforms. Wouldn't the appropriate place for responding to the unstable and deciding to stop the build if so desired be in the push installer workflow, rather than here?